### DISCUSSION TITLE
Welcome!

### DISCUSSION THREAD

**Discussion by jbcaillau** [control-toolbox/control-toolbox.github.io#13] [General]
@ocots @PierreMartinon Welcome to the `general` category of discussions in the ct: control-toolbox organisation. Can be a place for chatting (as discord / slack…) before melting things into an issue, *etc.* Accepts markdown with maths and more:

```math
g(t_f, x(t_f)) + \int_0^{t_f} f^0(x(t),u(t))\ \mathrm{d} t \to \min
```

**Comment by jbcaillau** [comment]
J'ai systématiquement cette erreur sur le package `HTTP`, vous avez déjà vu ça ?
![DAF639B6-0DB4-493D-9C05-38FCCFB3315B](https://user-images.githubusercontent.com/62183989/223494620-96e85903-28b5-4e99-be36-8d286302df3c.jpeg)

**Reply by ocots** [reply]
Tu peux faire `precompile` pour avoir plus d'infos et nous les partager. J'ai jamais eu cette erreur.

**Reply by ocots** [reply]
D'où vient ce paquet HTTP ? Il n'est pas dans le toml de `CTProblemLibrary`.

**Reply by jbcaillau** [reply]
pas gênant dans l'immédiat, je me penche sur la question plus tard. voilà en tout cas le precompile qui ne dit rien de plus. le package est présent dans mon Manifest comme dépendance à GR (backend graphique), visiblement (mais pas dans Project ce qui fait qu'un `remove` ne fait rien). bref
![5797B57F-5C2B-4424-849B-2D46EB623DA7](https://user-images.githubusercontent.com/62183989/223765287-c13bfa16-0112-4eb3-828c-8cd86c36479d.jpeg)
![995F995A-DB28-4ED0-8D2D-04BD4CFCC575](https://user-images.githubusercontent.com/62183989/223765300-26e09562-0dbc-4bec-b7a9-0fe537bc6cc7.jpeg)

**Comment by jbcaillau** [comment]
@ocots what about documentation generation in CTBase et al?

**Reply by ocots** [reply]
Could you formulate a more explicit question please?

**Reply by ocots** [reply]
https://control-toolbox.github.io/CTBase.jl/dev/api.html

**Comment by ocots** [comment]
Doc:
* https://documenter.juliadocs.org/stable/man/hosting/#gh-pages-Branch
* update
```yaml
  push:
    branches:
      - main
    tags: '*'
  pull_request:
```

**Reply by jbcaillau** [reply]
very good. did not remember it was already active.
- [ ] most functions in `ctparser_utils.jl` must not be exported (and shall not appear in the documentation)

**Reply by ocots** [reply]
We have to update this [file](https://github.com/control-toolbox/CTBase.jl/blob/main/docs/src/api.md) to export only what we want to be visible. But, the API is useful either for users and developers. So we need to structure a little bit the doc.

**Comment by jbcaillau** [comment]
@ocots ever tried this? maybe a bit redundant with github notifications + discussions + issues. also possible on slack.
[Github to Discord Webhook Tutorial](https://gist.github.com/SGTGunner/50d6a3cc0d489cf779f77695ba3e22ea)

**Reply by ocots** [reply]
Never tried. For me GitHub is enough. The only thing we don't have is visio. For the notifications, I like to be aware of it simply going to any GitHub page and not being spammed elsewhere.

**Comment by jbcaillau** [comment]
To check when github is down:

[Status](https://www.githubstatus.com)

**Comment by jbcaillau** [comment]
About `Documenter.jl`, I find the syntax for math sucks a bit (why not basic markdown + \$ for inline math...) A better idea? At least `math` works for displayed math...
![BA100118-810C-43CD-A19A-81D5A6F86122](https://user-images.githubusercontent.com/62183989/229616438-e1332394-1019-45fb-b018-f78ff11175fd.jpeg)
